addons/web/static/tests/mock_server_tests.js

```diff
@@ -94,3 +111,82 @@ QUnit.test("performRPC: search_read with active_test=true", async function (asse
     });
     assert.deepEqual(result, [{ id: 1, name: "Jean-Michel" }]);
 });
+
+QUnit.debug("performRPC: read_group, group by date", async function (assert) {
+    assert.expect(5);
+    const server = new MockServer(this.data, {});
+    let result = await server.performRPC("", {
+        model: "bar",
+        method: "read_group",
+        args: [[]],
+        kwargs: {
+            fields: ["foo"],
+            domain: [],
+            groupby: ["date"], //Month by default
+        },
+    });
+    assert.deepEqual(
+        result.map((x) => x.date),
+        ["December 2016", "October 2016", "April 2016"]
+    );
+
+    result = await server.performRPC("", {
+        model: "bar",
+        method: "read_group",
+        args: [[]],
+        kwargs: {
+            fields: ["foo"],
+            domain: [],
+            groupby: ["date:day"],
+        },
+    });
+    assert.deepEqual(
+        result.map((x) => x["date:day"]),
+        ["2016-12-14", "2016-10-26", "2016-12-15", "2016-04-11"]
```

Comment:
add an assert on count, and add a record to have at least one group with multiple records